aries_vcx/src/handlers/connection/invitee/state_machine.rs

```diff
@@ -354,46 +334,40 @@ impl SmConnectionInvitee {
             InviteeFullState::Completed(state) => {
                 InviteeFullState::Completed((state.clone(), disclose.protocols).into())
             }
-            _ => {
-                state.clone()
-            }
+            _ => state.clone()
         };
         Ok(Self { state, ..self })
     }
 
     pub fn handle_send_ack(self) -> VcxResult<Self> {
-        let Self { state, pairwise_info, send_message, .. } = self;
-        let state = match state {
-            InviteeFullState::Responded(state) => {
-                match Self::_send_ack(&state.did_doc, &state.request, &state.response, &pairwise_info, send_message) {
-                    Ok(response) => InviteeFullState::Completed((state, response).into()),
+        let state = match self.state {
+            InviteeFullState::Responded(ref state) => {
+                match self._send_ack(&state.did_doc, &state.response, &self.pairwise_info, self.send_message) {
+                    Ok(response) => InviteeFullState::Completed((state.clone(), response).into()),
                     Err(err) => {
                         let problem_report = ProblemReport::create()
                             .set_problem_code(ProblemCode::ResponseProcessingError)
                             .set_explain(err.to_string())
-                            .set_thread_id(&state.request.id.0);
-                        send_message(&pairwise_info.pw_vk, &state.did_doc, &problem_report.to_a2a_message()).ok();
-                        InviteeFullState::Initial((state, problem_report).into())
+                            .set_thread_id(&self.thread_id);
+                        (self.send_message)(&self.pairwise_info.pw_vk, &state.did_doc, &problem_report.to_a2a_message()).ok();
```

Comment:
nitpick - can't be without the parenthesis around `self.send_message`?

Reply:
No (complaints please send to the Rust compiler :))